### ISSUE TITLE
Allow custom headers on non-custom domains

### ISSUE THREAD

**Issue by ericholscher** [readthedocs/readthedocs.org#8649]
Currently we have the ability to set a custom HTTP header on custom domains, but we have had users who want to set custom headers on their existing `rtd.io` domain. 

## Implementation

Currently we have `HttpHeader`'s that link to a project's `Domain`:

https://github.com/readthedocs/readthedocs.org/blob/2552bb609ca46865dc36401dee0b1865a0aee52d/readthedocs/projects/models.py#L1596-L1600

It would make sense to actually have this relation point at the `Project` as well as the `Domain`.

## Complications

I'm guessing the reason that we have pointed this at the Domain is because of subprojects. So every project that is using that same Domain should use those custom headers. I think we want to be able to point it at either the Project *or* the Domain. This would allow this feature to be used by projects without a custom domain

## Use case

The user who pointed this out wanted add a CORS header that allowed CORS from their PR build domain. So, basically `Access-Control-Allow-Origin: docs--*.org.readthedocs.build`. Unfortunately, I'm not sure if CORS supports wildcards like this. Since the domain will change for each PR build, this might not allow us to set the CORS header manually, and it might need to be something that is specified in Python:

https://github.com/readthedocs/readthedocs.org/blob/d6f7347f82d7b3e210cd7e7e84d860282dd029f8/readthedocs/core/signals.py#L36

However, I still think it should be possible to add custom headers to projects without a custom domain, so this issue is worth fixing even if it doesn't work for this particular use case.

/cc @drammock

**Comment by humitos** [comment]
> The user who pointed this out wanted add a CORS header that allowed CORS from their PR build domain. So, basically `Access-Control-Allow-Origin: docs--*.org.readthedocs.build`. Unfortunately, I'm not sure if CORS supports wildcards like this

No, this is not supported by this header. However, this particular use case is implemented in https://github.com/readthedocs/readthedocs.org/pull/10737
